### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by RSydorenko322**
- [DEMO LINK](https://RSydorenko322.github.io/layout_stars/)
- [TEST REPORT LINK](https://RSydorenko322.github.io/layout_stars/report/html_report/)

- [x] Yellow stars are added with container modifier + pseudo-selector (NO extra classes)
- [x] Each BEM block has its own separate file
- [x] All `Typical Mistakes` from `BEM` lesson theory are checked.
- [x] Code follows all the [Code Style Rules ❗️](./checklist.md)

### CODE REVIEW

File: src/gray stars.css
Comment (lerastarynets) [inline src/gray stars.css:1]: such a strange name for file, pls move all styles to `style.css` file

File: src/stars--1.css
Comment (lerastarynets) [inline src/stars--1.css:1]: don't separate styles to different files like so, pls for now keep them all in style.css

File: src/style.css [hunk 1]
```diff
@@ -1 +1,118 @@
...
+
+/* #region stars--1 */
+.stars--1 .stars__star:nth-child(-n + 1) {
+  background-image: url(images/star-active.svg);
+  background-repeat: no-repeat;
+  background-position: center;
+  margin-left: 4px;
+  display: block;
+  width: 16px;
+  height: 16px;
```
Comment (anastasiiavorobiova) [inline src/style.css:34]:
```suggestion

```

File: src/style.css [hunk 2]
```diff
@@ -1 +1,118 @@
...
+  background-image: url(images/star.svg);
+  background-repeat: no-repeat;
+  background-position: center;
+  margin-left: 4px;
+}
+
+/* #endregion */
+
+/* #region stars--1 */
+.stars--1 .stars__star:nth-child(-n + 1) {
```
Comment (anastasiiavorobiova) [inline src/style.css:27]:
```suggestion
.stars--1 .stars__star:nth-child(-n + 1),
.stars--2 .stars__star:nth-child(-n + 2),
.... {
```
You can use commas for grouping selectors and avoid repeating code

File: src/style.css (hunk 1 above)
Comment (anastasiiavorobiova) [reply src/style.css:34]: No need to repeat the same styling everywhere. It's enough to have this in the basic class

File: src/style.css [hunk 3]
```diff
@@ -1 +1,118 @@
...
+  margin-left: 4px;
+  display: block;
+  width: 16px;
+  height: 16px;
+}
+
+/* #endregion */
+
+/* #region margins-zero */
+.stars--0 .stars__star:first-of-type {
```
Comment (anastasiiavorobiova) [inline src/style.css:94]:
```suggestion
.stars .stars__star:first-of-type {
```

File: src/style.css [hunk 4]
```diff
@@ -1 +1,118 @@
...
+  margin-left: 0;
+}
+
+.stars--4 .stars__star:first-of-type {
+  margin-left: 0;
+}
+
+.stars--5 .stars__star:first-of-type {
+  margin-left: 0;
+}
```
Comment (anastasiiavorobiova) [inline src/style.css:116]:
```suggestion

```

File: src/style.css [hunk 5]
```diff
@@ -1 +1,118 @@
...
+.stars--5 .stars__star:nth-child(-n + 5) {
+  background-image: url(images/star-active.svg);
+  background-repeat: no-repeat;
+  background-position: center;
+  margin-left: 4px;
+  display: block;
+  width: 16px;
+  height: 16px;
+}
+
```
Comment (anastasiiavorobiova) [inline src/style.css:90]:
```suggestion

```

File: src/style.css [hunk 6]
```diff
@@ -1 +1,118 @@
...
+/* #endregion */
```
Comment (anastasiiavorobiova) [inline src/style.css:50]:
```suggestion

```

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Almost done!

File: src/style.css [hunk 7]
```diff
@@ -1 +1,46 @@
...
+
+/* #region margins-zero */
+.stars--0 .stars__star:first-of-type,
+.stars--1 .stars__star:first-of-type,
+.stars--2 .stars__star:first-of-type,
+.stars--3 .stars__star:first-of-type,
+.stars--4 .stars__star:first-of-type,
+.stars--5 .stars__star:first-of-type {
+  margin-left: 0;
+}
```
Comment (anastasiiavorobiova) [inline src/style.css:44]:
```suggestion
.stars .stars__star:first-of-type {
  margin-left: 0;
}
```

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great progress here! There is the last thing to do

File: src/style.css [hunk 8]
```diff
@@ -1 +1,46 @@
...
+/* #endregion */
+
+/* #region margins-zero */
+.stars--0 .stars__star:first-of-type,
+.stars--1 .stars__star:first-of-type,
+.stars--2 .stars__star:first-of-type,
+.stars--3 .stars__star:first-of-type,
+.stars--4 .stars__star:first-of-type,
+.stars--5 .stars__star:first-of-type {
+  margin-left: 0;
```
Comment (Viktor-Kost) [inline src/style.css:43]: it's actually not a good way to reset value like this. 
You can do much easier

```
.stars__star:not(:first-child) {
  margin-left: 4px;
}
```

and remove margin here
```
.stars__star {
  display: block;
  width: 16px;
  height: 16px;
  background-image: url(images/star.svg);
  background-repeat: no-repeat;
  background-position: center;
  margin-left: 4px;
}
```

**Review by Viktor-Kost — APPROVED** [review]
Approved, yet please apply my comment

**Pull request closed without merging**